shared/utils/domains.ts

```diff
@@ -1,69 +1,48 @@
 import { trim } from "lodash";
+import env from "../env";
 
 type Domain = {
-  tld: string;
   subdomain: string;
   domain: string;
+  custom?: boolean;
 };
 
+// strips protocol and whitespace from input
+// then strips the path and query string
+function normalizeUrl(url: string) {
+  return trim(url.replace(/(https?:)?\/\//, "")).split(/[/:?]/)[0];
+}
+
 // we originally used the parse-domain npm module however this includes
 // a large list of possible TLD's which increase the size of the bundle
 // unnecessarily for our usecase of trusted input.
-export function parseDomain(url?: string): Domain | null | undefined {
-  if (typeof url !== "string") {
-    return null;
-  }
-  if (url === "") {
+export function parseDomain(url: string): Domain | null | undefined {
+  if (!url) {
     return null;
   }
 
-  // strip extermeties and whitespace from input
-  const normalizedDomain = trim(url.replace(/(https?:)?\/\//, ""));
-  const parts = normalizedDomain.split(".");
+  const normalBaseUrl = normalizeUrl(env.URL);
+  const normalUrl = normalizeUrl(url);
 
-  // ensure the last part only includes something that looks like a TLD
-  function cleanTLD(tld = "") {
-    return tld.split(/[/:?]/)[0];
+  const baseUrlStart = normalUrl.indexOf(`.${normalBaseUrl}`);
+  if (baseUrlStart === -1) {
+    return { subdomain: "", domain: normalUrl, custom: true };
```

Comment:
🍖 coupling the result of a parse with whether or not it is a custom domain. Otherwise, there was a lot of violation of [law of demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) going on whenever domains were being handled